core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/Variance.java

```diff
@@ -16,17 +16,28 @@
 
 package io.cdap.plugin.batch.aggregator.function;
 
+import com.google.common.collect.Table;
 import io.cdap.cdap.api.data.format.StructuredRecord;
 import io.cdap.cdap.api.data.schema.Schema;
 import io.cdap.plugin.batch.aggregator.AggregationUtils;
 
+import javax.annotation.Nullable;
+
 /**
- * Calculates Variance
+ * Calculates Sample Variance
+ * TODO: CDAP-16911 figure out a way to compute population variance
```

Comment:
I thought the difference between population and sample variance is just the denominator, where one divides by n, and the other by n+1? If you're keeping track of the squared mean and count, it seems like it should be straightforward division?

Reply:
Variance doesn't work in this way. The squared mean only makes sense for its own group since the mean and count are accumulated across the data. 
If you consider following code:
```
    count++;
    double delta = value.doubleValue() - mean;
    mean += delta / count;
    double delta2 = value.doubleValue() - mean;
    squareMean += delta * delta2;
```
For entire group, we cannot simply add up square mean as the new square mean because the actual square mean should be computed using the actual mean and count across the entire data.

Reply:
I mean, isn't the population variance just the sample variance * (N-1)/N, where N is the count?

Reply:
Ah, you are right, I thought it is complicated than this, will fix.